src/main/java/uk/gov/di/gpg45engine/services/impl/ProfileMatchingServiceImpl.java

```diff
@@ -0,0 +1,88 @@
+package uk.gov.di.gpg45engine.services.impl;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Service;
+import uk.gov.di.gpg45engine.domain.data.IdentityEvidence;
+import uk.gov.di.gpg45engine.domain.gpg45.EvidenceScore;
+import uk.gov.di.gpg45engine.domain.gpg45.IdentityProfile;
+import uk.gov.di.gpg45engine.matchers.ScoreMatcher;
+import uk.gov.di.gpg45engine.services.ProfileMatchingService;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Slf4j
+@Service
+public class ProfileMatchingServiceImpl implements ProfileMatchingService {
+
+    private final List<IdentityProfile> identityProfiles;
+
+    @Autowired
+    public ProfileMatchingServiceImpl(
+        @Qualifier("identity-profile-list") List<IdentityProfile> identityProfiles
+    ) {
+        this.identityProfiles = identityProfiles;
+    }
+
+    @Override
+    public IdentityProfile matchEvidenceScoringToProfile(IdentityEvidence[] evidenceBundle) {
+        var possibleMatches = new ArrayList<IdentityProfile>();
+
+        var evidenceScores = Arrays.stream(evidenceBundle)
+            .map(IdentityEvidence::getEvidenceScore)
+            .collect(Collectors.toList());
+
+        identityProfiles
+            .stream()
+            .takeWhile(_unused -> possibleMatches.size() != 1)
+            .forEach(identityProfile -> {
+
+                if (identityProfile.getEvidenceScoreCriteria().length > evidenceScores.size()) {
+                    log.debug("Skipped profile (criteria requires more evidence than provided): {}", identityProfile.getDescription());
+                    // skip due to lack of evidence provided.
+                    return;
+                }
+
+                // Compare the other 3 things to see if they match or not.
+
```

Comment:
We need to check the other 3 attributes as well, as the decision as to which profile is relevant may come down to the activityHistory, identityFraud or verification